src/rdsk.c

```diff
@@ -27,473 +27,800 @@
-			sprintf(status, "Unlocked");
-		} else {
-			sprintf(status, "Unavailable");
-		}
+	if ((fd = open(file, O_WRONLY)) < SUCCESS)
+		return -ENOENT;
 
-		printf(" RapidDisk Device %d: %s\tSize (KB): %llu\tUsage (KB): %llu\tStatus: %s\n", num,
-		       rd_prof->device, (rd_prof->size / 1024), (rd_prof->usage / 1024), status);
-		num++;
-		rd_prof = rd_prof->next;
-	}
-	printf("\nList of RapidDisk-Cache mapping(s):\n\n");
-	if (rc_prof == NULL) {
-		printf("  None\n");
-		goto list_out;
-	}
-	num = 1;
-	while (rc_prof != NULL) {
-		if (strstr(rc_prof->device, "rc-wb") != NULL) {
-			printf(" dm-writecache Target   %d: %s\tCache: %s  Target: %s (WRITEBACK)\n",
-				num, rc_prof->device, rc_prof->cache, rc_prof->source);
-		} else {
-			printf(" RapidDisk-Cache Target %d: %s\tCache: %s  Target: %s (%s)\n",
-				num, rc_prof->device, rc_prof->cache, rc_prof->source,
-				(strncmp(rc_prof->device, "rc-wt_", 5) == 0) ? \
-				"WRITE THROUGH" : "WRITE AROUND");
-		}
-		num++;
-		rc_prof = rc_prof->next;
+	if((ioctl(fd, BLKROGET, &rc)) == INVALID_VALUE) {
+		close(fd);
+		return -EIO;
 	}
-list_out:
-	printf("\n");
-	return SUCCESS;
-}
 
-int cache_device_stat(struct RC_PROFILE *rc_prof, unsigned char *cache)
-{
-	unsigned char cmd[NAMELEN] = {0};
+	close(fd);
 
-	if (rc_prof == NULL) {
-		printf("  No RapidDisk-Cache Mappings exist.\n\n");
-		return 1;
-	}
-	sprintf(cmd, "dmsetup status %s", cache);
-	system(cmd);
-	return SUCCESS;
+	return rc;
 }
 
-int cache_device_stat_json(struct RC_PROFILE *rc_prof, unsigned char *cache)
+/**
+ * It opens the device file, calls the ioctl() function to get the usage, and returns the result
+ *
+ * @param string The name of the device.
+ *
+ * @return The amount of memory used by the device.
+ */
+unsigned long long mem_device_get_usage(char *string)
 {
-	int rc = INVALID_VALUE;
-	unsigned char cmd[NAMELEN] = {0};
-	FILE *stream;
-	unsigned char *buf = NULL, *dup = NULL, *token = NULL;
-	struct RC_STATS *stats = NULL;
+	int fd;
+	unsigned long long rc = INVALID_VALUE;
+	char file[NAMELEN] = {0};
 
-	if (rc_prof == NULL) {
-		printf("  No RapidDisk-Cache Mappings exist.\n\n");
-		return 1;
+	sprintf(file, "/dev/%s", string);
+
+	if ((fd = open(file, O_WRONLY)) < SUCCESS)
+		return -ENOENT;
+
+	if ((ioctl(fd, RD_GET_USAGE, &rc)) == INVALID_VALUE) {
+		close(fd);
+		return -EIO;
 	}
+	close(fd);
+
+	return (rc * PAGE_SIZE);
+}
 
-	buf = (unsigned char *)calloc(1, BUFSZ);
-	if (buf == NULL) {
-		printf("%s: %s: calloc: %s\n", DAEMON, __func__, strerror(errno));
+/**
+ * It sends a flush command to the device mapper
+ *
+ * @param device the device name, e.g. /dev/mapper/my_device
+ *
+ * @return The return value is the return value of the dm_task_run function.
+ */
+int dm_flush_device(char *device) {
+	int cmdno = DM_DEVICE_TARGET_MSG;
+	struct dm_task *dmt;
+	uint64_t sector = 0;
+	char *command = "flush";
+
+	if (!(dmt = dm_task_create(cmdno))) {
+//		printf("%s\n", "dm_task_create error");
```

Comment:
We should clean up and/or remove the commented code here and below and in the other functions of this file.

Reply:
Done.